### PR TITLE
Sprint2 code refactoring and warning minimization

### PR DESCRIPTION
**Pull request by rinku-kumar-psl**
…sprint2

Signed-off-by: rinku-kumar-psl <rinku_kumar@persistent.com>

### Description
Sprint2 code refactoring and warning minimization

### Issues Resolved
[List any issues this PR will resolve]
#870 

Note: Internal PR review has been done here: https://github.com/abasatwar/observability/pull/52

### Check List
- [ ] Minimizes warning for all charts and refactored the code.
  - [ ] All tests pass, including unit test, integration test, and doctest
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
++ @spattnaik @abasatwar

### CODE REVIEW

File: dashboards-observability/common/types/explorer.ts [hunk 1]
```diff
@@ -247,4 +247,4 @@ export interface HistogramConfigList {
 export interface ConfigList {
   dimensions?: ConfigListEntry[] | HistogramConfigList[];
   metrics?: ConfigListEntry[];
-}
+}
```
Comment (abasatwar) [inline dashboards-observability/common/types/explorer.ts:250]: add empty line at end.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 2]
```diff
@@ -301,4 +301,4 @@ export const ConfigPanel = ({ visualizations, setCurVisId, callback, changeIsVal
       </EuiFlexGroup>
     </>
   );
-};
+};
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:304]: same as above.

File: dashboards-observability/public/components/visualizations/charts/pie/pie.tsx [hunk 3]
```diff
@@ -17,42 +17,42 @@ export const Pie = ({ visualizations, layout, config }: any) => {
...
-    return dimLabels.map((label:string, index:number) => {
+  const createLegendLabels = (dimLabels: string[], xaxisLables: string[]) => {
+    return dimLabels.map((label: string, index: number) => {
       return [xaxisLables[index], label].join(',');
     });
   };
-  
-  const labelsOfXAxis = () => {
+
+  const labelsOfXAxis = useMemo(() => {
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/pie/pie.tsx:55]: do we need useMemo here.

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/visualizations/charts/pie/pie.tsx (hunk 3 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/visualizations/charts/pie/pie.tsx:55]: As we have utilized the createLegendLabels method to map Legend labels. That should be updated only on data and axis change. So we have created memoize value here.

**Review by spattnaik — APPROVED** [review]

**Review by abasatwar — APPROVED** [review]

**Review by anirudha — APPROVED** [review]

**Pull request merged**